tests/unittest/_torch/attention/sparse/test_cute_dsl_fp4_paged_mqa_logits.py

```
@@ -0,0 +1,495 @@
+    block_tables: torch.Tensor,
+    max_model_len: int,
+):
+    """Pure PyTorch reference for paged MQA logits.
+
+    Inputs are already in the simulated dtype (after FP4 quant->dequant
+    cast back). Body mirrors DeepGEMM's ``ref_paged_mqa_logits``: per-batch
+    MQA matmul -> causal/context mask -> ReLU -> weighted sum across heads.
+    Returns logits in float32; the kernel output is cast to float for
+    comparison.
+    """
+    batch_size, next_n, num_heads, dim = q.size()
```

Ruff (0.15.12)

[warning] Unpacked variable `num_heads` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)